js/kucoinfutures.js

```diff
@@ -1059,11 +1064,32 @@ module.exports = class kucoinfutures extends kucoin {
             'leverage': 1,
         };
         const stopPrice = this.safeValue2 (params, 'triggerPrice', 'stopPrice');
+        const stopLossPrice = this.safeValue (params, 'stopLossPrice');
+        const takeProfitPrice = this.safeValue (params, 'takeProfitPrice');
+        const isStopLoss = stopLossPrice !== undefined;
+        const isTakeProfit = takeProfitPrice !== undefined;
+        const stop = this.safeString (params, 'stop');
+        const stopPriceType = this.safeString (params, 'stopPriceType', 'MP');
         if (stopPrice) {
-            request['stop'] = (side === 'buy') ? 'up' : 'down';
-            const stopPriceType = this.safeString (params, 'stopPriceType', 'TP');
-            request['stopPriceType'] = stopPriceType;
+            if (stop === undefined) {
+                request['stop'] = (side === 'buy') ? 'down' : 'up';
```

Comment:
These are backwards, when you set a stop buy order, it's for a short, you're saying that when the price rises above (up) past a certain price, then buy the currency, which will exit the short. I already fixed this in https://github.com/ccxt/ccxt/pull/16619

Reply:
I think this makes sense for stop orders, I don't think we should change stopLoss and takeProfits though